src/Api.js

```diff
@@ -0,0 +1,82 @@
+import camelCase from 'lodash.camelcase';
+import snakeCase from 'lodash.snakecase';
+
+import App from './App';
+
+export const modifyObjectKeys = (object, modify) => {
+  // If the passed in object is not an Object, return it.
+  if (
+    object === undefined ||
+    object === null ||
+    (typeof object !== 'object' && !Array.isArray(object))
+  ) {
+    return object;
+  }
+
+  if (Array.isArray(object)) {
+    return object.map(value => modifyObjectKeys(value, modify));
+  }
+
+  // Otherwise, process all its keys.
+  const result = {};
+  Object.entries(object).forEach(([key, value]) => {
+    result[modify(key)] = modifyObjectKeys(value, modify);
+  });
+  return result;
+};
+
+export const camelCaseObject = object => modifyObjectKeys(object, camelCase);
+
+export const snakeCaseObject = object => modifyObjectKeys(object, snakeCase);
+
+export const convertKeyNames = (object, nameMap) => {
+  const transformer = key =>
+    (nameMap[key] === undefined ? key : nameMap[key]);
+
+  return modifyObjectKeys(object, transformer);
+};
+
+export function formatKnownError(error) {
+  const formattedError = new Error();
+  const { fieldErrors, errors, messages } = camelCaseObject(error.response.data);
+  formattedError.fieldErrors = fieldErrors;
+  formattedError.errors = errors;
+  formattedError.messages = messages;
+  return formattedError;
+}
+
+export function isKnownError(error) {
+  return (
+    error.response !== undefined &&
+    error.response.data !== undefined &&
+    (
+      error.response.data.errors !== undefined ||
+      error.response.data.field_errors !== undefined ||
+      error.response.data.messages !== undefined
+    )
+  );
+}
+
+export const request = async (type, ...args) => {
```

Comment:
Could we leverage axios interceptors instead?

Reply:
This code is being removed - we decided not to include it since we're not convinced it's the right shape.